### PR TITLE
New MapLoader

### PR DESCRIPTION
**Pull request by ArchDemons**

### CODE REVIEW

**Comment by tonihele** [comment]
A lot to take in :D But overall seems like a job well done, thanks. However, I found your lack of water & lava and floating door locks disturbing... I just gave it a quick spin. The cinematic seems to be changed a bit? Visually I mean, the camera is a bit lower? I don't know if this is a fault or not, just commenting, or am I imagining this...?

Did this fix some of the room builds? At least the temple in one of the campaign levels looked better than before.

The door locks I remember were really tricky :( Maybe you fixed something and now my hacks wont work.

**Comment by ArchDemons** [comment]
Today I also found bug with water and lava. Tomorrow push fixes. 
Because map was higher on 1f, camera was lower on 1f. Maybe camera settings change it back (we still do not know how to configure it properly).
I fix DoubleQuad constructor and all rooms on it. 
I don`t check  door locks. Check it tomorrow. Problem maybe with lock height position as water and lava.

**Comment by ArchDemons** [comment]
I fixed all what found

File: src/toniarts/openkeeper/tools/convert/map/KwdFile.java [hunk 1]
```diff
@@ -380,7 +381,7 @@ private void readFileContents(KwdHeader header, RandomAccessFile data) throws IO
     }
 
     /**
-     * Reads the *Map.kld
+     * Reads the *GameMap.kld
```
Comment (tonihele) [inline src/toniarts/openkeeper/tools/convert/map/KwdFile.java]: No :) You search & replaced too much :)

File: src/toniarts/openkeeper/tools/convert/map/KwdFile.java [hunk 2]
```diff
@@ -3167,7 +3168,7 @@ private void checkOffset(long itemSize, RandomAccessFile file, long offset) thro
         private int checkThree; // in level = 223
         private int dataSizeLevel; // in Level size of data exclude paths
         private int unknown; // only in Triggers and Level
-        private int headerEndOffset; // 28, *Map - 8, *Triggers - 32,
+        private int headerEndOffset; // 28, *GameMap - 8, *Triggers - 32,
```
Comment (ufdada) [inline src/toniarts/openkeeper/tools/convert/map/KwdFile.java]: This shouldn't be renamed, too :)

**Comment by tonihele** [comment]
Nice work sir! I found a small bug, but I'll just fix it after the merge.

**Comment by tonihele** [comment]
Oh and a nice to see the door somewhat working :)

**Comment by tonihele** [comment]
Indeed the camera is too low. You can see it from the creature entrance video. I'm not going to touch it, but you can fix it if you want. I have a lot to do now, putting all this back to #263 :)

**Comment by ArchDemons** [comment]
ok

**Comment by ArchDemons** [comment]
I think we should stay this changes as is. We should not roll back this changes, until we understand what wrong: camera height or map height. Why? If you see torches height in rooms (in editor), you see 0.5f or 1.5f. In current state all good, but if we move map on y=-1f torch position not will not match

**Comment by tonihele** [comment]
Might also be things like origin of the models, or origin on the screen coordinates on view port (can't be...? we are not using any such thing?). Since I think we are converting from DirectX to OpenGL

**Comment by tonihele** [comment]
And the change is good as it is, and also the fact that I'm reaallllly trying to redesign our architecture now, so I appreciate that there would be no super major changes that are hard to pick to new architecture (and might even be not relevant anymore). I'll try to be quick with it, rather than try to implement a lot of new stuff at the same time, like I already actually tried (player maps, but our current map rendering is actually quite difficult, more specifically how we handle the rooms, the map loader needs to know too much just for rendering purposes)...

**Pull request merged**